saviSimulation/savi/jason_processing/behaviour/UxVBehavior.java

```diff
@@ -0,0 +1,240 @@
+package savi.jason_processing.behaviour;
+
+import java.util.*;
+
+import processing.core.*;
+
+import savi.StateSynchronization.*;
+import savi.jason_processing.AgentModel;
+import savi.jason_processing.Geometry;
+import savi.jason_processing.UxV;
+import savi.jason_processing.WorldObject;
+
+
+public abstract class UxVBehavior extends AgentModel {
+    private static final float SPEED = 0.1f; // 0.1 pixels (whatever real-life distance this corresponds to)
+
+    //-----------------------------------------
+    // DATA (or state variables)
+    //-----------------------------------------
+    //String ID; -- Note: moved to superclass
+    //String type; -- same
+    //SyncAgentState agentState; --same
+    protected PVector initialPosition;
+    protected PVector speedVector;
+    protected double compasAngle;
+    protected ArrayList<CameraPerception> visibleItems;
+    protected double time;
+    protected Map<String, AgentAction> actionMap;
+
+    //***********************************************************//
+    //I THINK IS BETTER TO HAVE THE ROBOTS ITS DATA AND THE SYNCAGENTSTATE ITS OWN.
+    //IF WE WANT TO IMPLEMENTE MALFUNCTION OF SENSORS, THE INFO RECEIVED IN
+    //SYNCAGENTSTATE IS NOT THE REAL ONE
+    //***********************************************************//
+
+    //-----------------------------------------
+    // METHODS (functions that act on the data)
+    //-----------------------------------------
+
+    /**
+     * Constructor
+     *
+     * @param id
+     * @param type
+     * @param initialPosition
+     */
+    public UxVBehavior(String id, String type, PVector initialPosition, double reasoningCyclePeriod) {
+        // Initialize data values
+        super(reasoningCyclePeriod);
+        this.ID = id;
+        this.type = type;
+        this.initialPosition = initialPosition;
+        this.speedVector = new PVector();
+        this.time = 0;
+        this.compasAngle = 0;
+        this.agentState = new SyncAgentState();
+        this.visibleItems = new ArrayList<CameraPerception>();
+        this.actionMap = new HashMap<>();
```

Comment:
maybe some type parameters could help clarify? it's String->AgentAction right?

Reply:
Typically these are not included and are considered redundant when initializing since the declaration defines the generics being used (line 28 of this file). 

If you think that it would still be good to have this redundancy, let me know and I will put them in! I have no issues with doing this either way.

Reply:
ok
I'm for the redundancy though, just for readability

Reply:
Cool with me! Done.